consensus/polybft/validators_snapshot.go

```diff
@@ -38,58 +36,55 @@ func (v *validatorsSnapshotCache) GetSnapshot(blockNumber uint64, parents []*typ
 	v.lock.Lock()
 	defer v.lock.Unlock()
 
-	epochNumber := getEpochNumber(blockNumber, v.epochSize)
-	if !isEndOfPeriod(blockNumber, v.epochSize) && epochNumber > 0 {
-		// snapshot is retrieved for the previous epoch
-		// since the last block of the previous epoch contains validator set for the current epoch
-		epochNumber--
-	}
-
-	v.logger.Trace("Retrieving snapshot started...", "Block", blockNumber, "Epoch", epochNumber)
+	v.logger.Trace("Retrieving snapshot started...", "Block", blockNumber)
 
 	var snapshot AccountSet
 
-	currentEpochNumber := epochNumber
+	lastBlockWithSnapshot := blockNumber
```

Comment:
We know here epoch number. Could we store validator set only for epochs and dont clean them?

Reply:
Is it worth to make https://polygon.atlassian.net/browse/EVM-155 there?

Reply:
See https://github.com/0xPolygon/polygon-edge/pull/897#discussion_r1021137452. 

I think we could store them by hash. I've taken a look at the code, and we are aware of the block extra on each point where we call `GetValidators` function. Let's discuss on sync. Maybe this is a good idea to implement on another PR.